### PR TITLE
fix: parallelize preprocess drop blob fix cold start preprocess

### LINKED ISSUE #1245: fix: text preprocessing has a cold start issue
The issue is nltk download which happens on the first request; we can solve this by now by moving it into the executor startup to avoid polluting the performance measurements.

### LINKED ISSUE #1246: fix: parallelise preprocessing as it's currently mainly I/O-bound
Downloading files takes the longest time in preprocessing images. Downloading them in a `ThreadPoolExecutor` greatly reduces preprocessing time.

### LINKED ISSUE #1247: fix: drop blob when we can
We don't drop blob when we have a URI anymore due to a bug in `aggregate_embeddings`; returning blob makes HTTP connections very slow

### PR DESCRIPTION
**Pull request by joschkabraun**
Closes #1245, closes #1246 and closes #1247.

---

- [x] This PR references an open issue
- [ ] Added a test
- [ ] Added a screenshot of a successful customer deployment

### CODE REVIEW

File: now/executor/abstract/auth/auth.py [hunk 1]
```diff
@@ -69,14 +70,18 @@ def wrapper(*args, **kwargs):
                 ]
                 cls_instance.logger.info(f"lengths: {', '.join(lengths)}")
 
+            t0 = time.time()
             res = func(*args, **kwargs)
+            t1 = time.time()
 
             if isinstance(res, DocumentArray):
-                cls_instance.logger.info(f"output {len(res)} docs")
+                cls_instance.logger.info(
```
Comment (makram93) [inline now/executor/abstract/auth/auth.py:78]: do we also have `docs_map` as one of the returned instances?

**Review by makram93 — APPROVED** [review]
LGTM! 🚀

File: now/app/base/app.py [hunk 2]
```diff
@@ -204,31 +205,25 @@ def setup(self, user_input: UserInput, testing=False, **kwargs) -> Dict:
     def preprocess(
         self,
         docs: DocumentArray,
+        max_workers: int = 1,
```
[the remark this answers is not in the document]
Comment (joschkabraun) [reply now/app/base/app.py:208]: I left it at 1 to mimic the sequential behaviour as before and for the unit tests. The preprocessor which mainly calls the function gives the function its own `max_workers`.

**Review by IyadhKhalfallah — APPROVED** [review]
Looks much more efficient, great work!

**Pull request merged**